src/graph/graphcontext.h

```diff
@@ -29,34 +29,35 @@ GraphContext* GraphContext_Retrieve(RedisModuleCtx *ctx, const char *graphname);
-LabelStore* GraphContext_AllStore(const GraphContext *gc, LabelStoreType t);
-// Retrieve the specific store for the provided node label or relation type string
-LabelStore* GraphContext_GetStore(const GraphContext *gc, const char *label, LabelStoreType t);
-// Add a new store and matrix for the given node label
-LabelStore* GraphContext_AddLabel(GraphContext *gc, const char *label);
-// Add a new store and matrix for the given relation type 
-LabelStore* GraphContext_AddRelationType(GraphContext *gc, const char *label);
+/* Schema API */
+// Retrieve the generic schema for node labels or relation types
+Schema* GraphContext_GetUnifiedSchema(const GraphContext *gc, SchemaType t);
+// Retrieve number of schemas created for given type.
+unsigned short GraphContext_SchemaCount(const GraphContext *gc, SchemaType t);
+// Retrieve the specific schema for the provided node label or relation type string
+Schema* GraphContext_GetSchema(const GraphContext *gc, const char *label, SchemaType t);
+// Retrieve the specific schema for the provided id.
+Schema* GraphContext_GetSchemaByID(const GraphContext *gc, int id, SchemaType t);
+// Add a new schema and matrix for the given label
+Schema* GraphContext_AddSchema(GraphContext *gc, const char *label, SchemaType t);
 
 /* Index API */
 bool GraphContext_HasIndices(GraphContext *gc);
-// Attempt to retrieve an index on the given label and property
-Index* GraphContext_GetIndex(const GraphContext *gc, const char *label, const char *property);
-// Create and populate an index for the given label and property
-int GraphContext_AddIndex(GraphContext *gc, const char *label, const char *property);
+// Attempt to retrieve an index on the given label and attribute
+Index* GraphContext_GetIndex(const GraphContext *gc, const char *label, const char *attribute);
+// Create and populate an index for the given label and attribute
+int GraphContext_AddIndex(GraphContext *gc, const char *label, const char *attribute);
 // Remove and free an index
-int GraphContext_DeleteIndex(GraphContext *gc, const char *label, const char *property);
+int GraphContext_DeleteIndex(GraphContext *gc, const char *label, const char *attribute);
 
 // Add a single node to all indices its properties match
-void GraphContext_AddNodeToIndices(GraphContext *gc, LabelStore *store, Node *n);
+void GraphContext_AddNodeToIndices(GraphContext *gc, Schema *s, Node *n);
 // Remove a single node from all indices that refer to it
-void GraphContext_DeleteNodeFromIndices(GraphContext *gc, LabelStore *store, Node *n);
+void GraphContext_DeleteNodeFromIndices(GraphContext *gc, Node *n);
 // Remove a single node property from an index and re-insert with new value
-void GraphContext_UpdateNodeIndices(GraphContext *gc, LabelStore *store, NodeID id, EntityProperty *prop, SIValue *newval);
+void GraphContext_UpdateNodeIndices(GraphContext *gc, Schema *s, NodeID id, Attribute_ID attid, SIValue *oldval, SIValue *newval);
```

Comment:
Function removed; delete declaration. Incidentally, I think the op_update code is much cleaner without this function! Nicely done.